pipeline/import/utils/defineEnvSource.R

```diff
@@ -56,11 +57,33 @@ if (dataSource == "geonorge") {
     
 ### 4. MODIS ####    
 } else if (dataSource == "modis") {
-  rasterisedVersion <- get_modis(regionGeometry, projCRS, focalParameter)
+  rasterisedVersion <- get_modis(regionGeometryBuffer, projCRS, focalParameter)
 } else if (dataSource == "corine") {
-  rasterisedVersion <- get_corine()
+  # check if encompassing corine alreadydownloaded
+  rasterisedVersion <- checkAndImportRast("land_cover_corine", regionGeometryBuffer, dataPath)
+  # download and save if missing
+  if(is.null(rasterisedVersion)){
+    # download
+    rasterisedVersion <- get_corine()  
+    # save
+    file_path <- generateRastFileName(rasterisedVersion, focalParameter, dataPath)
+    writeRaster(rasterisedVersion, filename = file_path, overwrite = TRUE)
+  }
+  # calculate distance to water (if necessary)
+  if (focalParameter == 'distance_water') {
+    # Extract the levels data frame
+    levels <- levels(rasterisedVersion)[[1]]
+    # Find the numeric value corresponding to "Water bodies"
+    water_val <- levels[levels$LABEL3 == "Water bodies", 1]
+    water <- rasterisedVersion == water_val
```

Comment:
There's a problem in this line, cause by the line above. My version of the raster doesn't have the column "LABEL3" in its 'levels' data frame, instead it's called "U2018_CLC2018_V2020_20u1". We could select the column name by asking which column has the class "character" instead?